website/content/docs/api-gateway/configuration/gateway.mdx

```diff
@@ -0,0 +1,324 @@
+Specifies an array of requirements for matching namespaces. If a match is found, then routes from the matching namespace(s) are allowed to attach to the `Gateway`. The following table describes members of the `matchExpressions` array:
+
+| Requirement | Description | Type | Required |
+|--- |--- |--- |--- |
+|`key` | Specifies the label that the `key` applies to. | string | required when `matchExpressions` is declared |
+|`operator` | Specifies the key's relation to a set of values. You can use the following keywords: <ul><li>`In`: Only routes in namespaces that contain the strings in the `values` field can attach to the `Gateway`. </li><li>`NotIn`: Routes in namespaces that do not contain the strings in the `values` field can attach to the `Gateway`. </li><li>`Exists`: Routes in namespaces that contain the `key` value are allowed to attach to the `Gateway`.</li><li>`DoesNotExist`: Routes in namespaces that do not contain the `key` value are allowed to attach to the `Gateway`.</li></ul> | string | required when `matchExpressions` is declared |
+|`values` | Specifies an array of string values. If `operator` is configured to `In` or `NotIn`, then the `values` array must contain values. If `operator` is configured to `Exists` or `DoesNotExist`, then the `values` array must be empty. This array is replaced during a strategic merge patch. | array of strings | required when `matchExpressions` is declared |
+
+In the following example, routes in namespaces that contain `foo` and `bar` are allowed to attach routes to the `Gateway`.
+```yaml
+namespaceSelector:
+  matchExpressions:
+  - key: kubernetes.io/metadata.name
+    operator: In
+    values:
+    - foo
+    - bar
+```
+
+### listeners.allowedRoutes.namespaces.selector.matchLabels
+Specifies an array of labels and label values. If a match is found, then routes with the matching label(s) are allowed to attach to the `Gateway`. This selector can contain any arbitrary key/value pair.
+
+In the following example, routes in namespaces that have a `bar` label are allowed to attach to the `Gateway`.
+
+```yaml
+namespaceSelector:
+    matchLabels:
+      foo: bar
+```
+
+Refer to [Labels and Selectors](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) in the Kubernetes documentation for additional information about labels.
+
+### listeners.hostname
+Specifies the `listener`'s hostname.
+* Type: string
+* Required: required
+
+### listeners.name
+Specifies the `listener`'s name.
+* Type: string
+* Required: required
+
+### listeners.port
+Specifies the port number that the `listener` attaches to.
+* Type: integer
+* Required: required
+
+### listeners.protocol
+Specifies the protocol the `listener` communicates on.
+* Type: string
+* Required: required
+
+Allowed values are `TCP`, `HTTP`, or `HTTPS`
+
+### listeners.tls
+Specifies the `tls` configurations for the `Gateway`. The `tls` object is required if `protocol` is set to `HTTPS`. The object contains the following fields:
```

Comment:
@im2nguyen - the allowedRoutes section has several levels of children and isn't conducive to being formatted in the same way. Tables are nicer, but the information they describe has to be relatively flat. We went with a table in this case because the children node only has one dependent that can easily appear in the table. I don't think you can do that with nodes higher in the configuration model.